src/execution_plan/ops/op_merge.c

```diff
@@ -54,7 +54,7 @@ static void _UpdateProperties(OpMerge *op, Record *records) {
 
 	// Lock everything.
 	if(!QueryCtx_LockForCommit()) return;
```

Comment:
There's room for improvement here, as Merge isn't a writer and shouldn't need to lock unless it has an `on_match` member. This would require the same check to be made in `_ExecutionPlan_FindLastWriteOp`.

Reply:
(I guess I wrote this, so my bad!)

Reply:
Leaving this as a "won't fix" for now.